### ISSUE TITLE
Make resource references clickable

### ISSUE THREAD

**Issue by Cooldude2606** [clusterio/clusterio#572]
tl;dr Change resources references labels into clickable elements that navigate you to the page for that resource. (Example resources: host, instance, user, role, etc)

### Motivation
There are many places where the name/id of a resource is displayed but in order to view that resource you must navigate to the list of all resources of that type and then click the one you want. This is slow and unintuitive and might only get worse as more plugins are developed.

### Solution
Anywhere that a resource reference is displayed on the web ui should be made clickable with a href or otherwise. It should be obvious to the user that the elements are clickable but are not buttons. This could mean the text is underlined and is highlighted on hover, or something like a chip style that google uses, or a translucent box with a chain icon as used by discord.
```[tasklist]
### Tasks
- [x] Host assign box on instance page
- [x] Host in instance list
- [x] User role tags in users table -> Role settings page
```

**Comment by Hornwitser** [comment]
Note that the way to create links with react-router and antd is something like this
 ```jsx
import { Typography } from "antd";
import { Link } from "react-router-dom";

// in code
<Link to="/instances/10/view" component={Typography.Link}>Instance 10</Link>
```

**Comment by Danielv123** [comment]
I could only find 2 instances of non interactive resources. If there are any more you can find, could you please specify?
